> [Fatal Error] sleep.c@23,10: 'bb_api.h' file not found
[ERROR] ./targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/sleep.c:23:10: fatal error: 'bb_api.h' file not found
#include "bb_api.h"

This build failing is also with the old tools. You can find it in the logs for greentea tests Let's fix it there first so they can see this bb_api.h file. Then CMake (we can help definitely. I would remove for now 32e1eaac36d09c391df5779636df32cabbd7fb36. If the 
 target depends on cordio stack, it should link to it it via target_link_libraries() ).

--- a/connectivity/drivers/ble/FEATURE_BLE/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/CMakeLists.txt
+++ b/connectivity/drivers/ble/FEATURE_BLE/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/CMakeLists.txt
@@ -17,6 +17,7 @@ target_sources(mbed-ble-cordio
         stack/sources/pal_bb_ble_rf.c
         stack/sources/pal_cfg.c
         stack/sources/pal_crypto.c
+        stack/sources/pal_rtc.c
         stack/sources/pal_timer.c
 )
 

--- a/connectivity/drivers/ble/FEATURE_BLE/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/NRFCordioHCIDriver.cpp
+++ b/connectivity/drivers/ble/FEATURE_BLE/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/NRFCordioHCIDriver.cpp
@@ -260,7 +260,13 @@ void NRFCordioHCIDriver::do_initialize()
     {
     }
 
-
+    if(NRF_CLOCK->LFCLKSRC == CLOCK_LFCLKSRC_SRC_RC) {
+        NRF_CLOCK->EVENTS_DONE = 0;
+        NRF_CLOCK->TASKS_CAL = 1;
+        while (NRF_CLOCK->EVENTS_DONE == 0)
+        {
+        }
+    }
 
     // mbed-os target uses IRQ Handler names with _v added at the end
     // (TIMER0_IRQHandler_v and TIMER2_IRQHandler_v) so we need to register these manually

--- a/connectivity/drivers/ble/FEATURE_BLE/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/stack/sources/pal_rtc.c
+++ b/connectivity/drivers/ble/FEATURE_BLE/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/stack/sources/pal_rtc.c
@@ -231,9 +231,6 @@ void PalRtcInit(void)
         PalRtcDisableCompareIrq(channelId);
     }
 
-    /* Configure low-frequency clock. */
-    NRF_CLOCK->LFCLKSRC             = (CLOCK_LFCLKSRC_SRC_Xtal << CLOCK_LFCLKSRC_SRC_Pos);
-
     NRF_CLOCK->EVENTS_LFCLKSTARTED  = 0;
     NRF_CLOCK->TASKS_LFCLKSTART     = 1;
     while (NRF_CLOCK->EVENTS_LFCLKSTARTED == 0) { }

--- a/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/common_rtc.c
+++ b/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/common_rtc.c
@@ -100,9 +100,6 @@ void COMMON_RTC_IRQ_HANDLER(void)
 __STATIC_INLINE void errata_20(void)
 {
 #if defined(NRF52_PAN_20)
-    if (COMMON_RTC_INSTANCE != NRF_RTC1) {
-        return;
-    }
     if (!NRF_HAL_SD_IS_ENABLED())
     {
         NRF_CLOCK->EVENTS_LFCLKSTARTED = 0;
@@ -112,7 +109,7 @@ __STATIC_INLINE void errata_20(void)
         {
         }
     }
-    NRF_RTC1->TASKS_STOP = 0;
+    COMMON_RTC_INSTANCE->TASKS_STOP = 0;
 #endif
 }
 

